src/FormField/DatePickerField.tsx

```diff
@@ -1,41 +1,47 @@
 import * as React from "react";
-import { props, t, ReactChild, ObjectOmit } from "../utils";
+import { ObjectOmit } from "../utils";
 import * as cx from "classnames";
-import View from "react-flexview";
 import DatePicker from "../DatePicker";
 import { FormField } from "./FormField";
 
+type DefaultProps<T extends DatePicker.Value> = {
+  /** An optional custom renderer for DatePicker */
+  datePickerRenderer: (props: DatePicker.Props<T>) => JSX.Element;
+};
+
+type NonDefaultProps<T extends DatePicker.Value> = {
+  /** the label for the field */
+  label: FormField.Props["label"];
+  /** whether the field is required */
+  required?: FormField.Props["required"];
+  /** optional props to pass to the wrapping View */
+  viewProps?: FormField.Props["viewProps"];
+  /** an optional hint describing what's the expected value for the field (e.g. sample value or short description) */
+  hint?: FormField.Props["hint"];
+  /** an optional class name to pass to top level element of the component */
+  className?: string;
+  /** an optional style object to pass to top level element of the component */
+  style?: React.CSSProperties;
+} & ObjectOmit<
+  DatePicker.Props<T>,
+  "returnFormat"
+>; /* returnFormat is causing issues with the inference of OnChangeProps */
```

Comment:
@gabro is proposing to remove `returnFormat` (and `T`s other than `Date`) completely

Reply:
https://github.com/buildo/react-components/issues/1347